doc/Implementing_a_provider.md

```diff
@@ -56,7 +56,10 @@ The ContentManager has the following methods:
  * `removeContentForUsers(string $appId, string $providerId, string $itemId, array $users)` Remove a content item from the knowledge base of context chat for specified users
  * `removeAllContentForUsers(string $appId, string $providerId, array $users)` Remove all content items from the knowledge base of context chat for specified users
 
-To register your content provider, call `registerContentProvider` in your app's Application#register() step.
+To register your content provider, your app needs to listen to the `OCA\ContextChat\Event\ContentProviderRegisterEvent` event and call the `registerContentProvider` method in the event for every provider you want to register.
+Any interaction with the content manager using the above listed methods should automatically register the provider.
+
+You may call the `registerContentProvider` method explicitly if you want to trigger an initial import of content items.
```

Comment:
Why would this be necessary?

Reply:
in reference to line 62, I assume. In cases where an app would just want to have all the existing data indexed and not wait for any data point to appear to call the `submitContent` method which would then register the provider and import the existing data with the initial import. In essence, implementing apps would maybe have an on/off option to index documents to context chat and upon turning it on, the register function is called, as one scenario.

Reply:
Ah, good point. We do also have the hook for when an app is enabled, right?

Reply:
no, only for app disabled, but excellent point :rocket:

Reply:
But we'd probably need another event for that, right? maybe let's not do that for now and leave it for apps to figure out when they want to trigger an initial import

Reply:
cool. Ready for merge?